gql/response_parser.py

```diff
@@ -0,0 +1,114 @@
+from typing import Any, Dict, Callable, Optional, List
+
+from graphql.type.schema import GraphQLSchema
+from graphql.type.definition import GraphQLObjectType, GraphQLField, GraphQLScalarType
+
+
+class ResponseParser(object):
+    """The challenge is to substitute custom scalars in a GQL response with their
+    decoded counterparts.
```

Comment:
This should be written more clearly: explain briefly what "custom scalars" are, and what "decoded" means in this context.